dask/dataframe/io/tests/test_parquet.py

```diff
@@ -4642,11 +4642,31 @@ def test_pyarrow_filesystem_option(tmp_path, fs):
 def test_select_filtered_column(tmp_path, engine):
     df = pd.DataFrame({"a": range(10), "b": ["cat"] * 10})
     path = tmp_path / "test_select_filtered_column.parquet"
-    df.to_parquet(path, index=False)
+    stats = {"write_statistics" if engine == "pyarrow" else "stats": True}
+    df.to_parquet(path, engine=engine, index=False, **stats)
 
     with pytest.warns(UserWarning, match="Sorted columns detected"):
+        ddf = dd.read_parquet(path, engine="pyarrow", filters=[("b", "==", "cat")])
```

Comment:
This now makes sure that the test fails with a meaningful error message when pyarrow is not installed:

```
  860s] tmp_path = PosixPath('/tmp/pytest-of-abuild/pytest-0/popen-gw1/test_select_filtered_column_fa3')
[  860s] engine = 'fastparquet'
[  860s] 
[  860s]     def test_select_filtered_column(tmp_path, engine):
[  860s]         df = pd.DataFrame({"a": range(10), "b": ["cat"] * 10})
[  860s]         path = tmp_path / "test_select_filtered_column.parquet"
[  860s]         stats = {"write_statistics" if engine == "pyarrow" else "stats": True}
[  860s]         df.to_parquet(path, engine=engine, index=False, **stats)
[  860s]     
[  860s]         with pytest.warns(UserWarning, match="Sorted columns detected"):
[  860s] >           ddf = dd.read_parquet(path, engine="pyarrow", filters=[("b", "==", "cat")])
[  860s] 
[  860s] /usr/lib/python3.10/site-packages/dask/dataframe/io/tests/test_parquet.py:4631: 
[  860s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[  860s] 
[  860s] args = (PosixPath('/tmp/pytest-of-abuild/pytest-0/popen-gw1/test_select_filtered_column_fa3/test_select_filtered_column.parquet'),)
[  860s] kwargs = {'engine': 'pyarrow', 'filters': [('b', '==', 'cat')]}
[  860s] func = <function read_parquet at 0xe128c028>
[  860s] 
[  860s]     @wraps(fn)
[  860s]     def wrapper(*args, **kwargs):
[  860s]         func = getattr(self, dispatch_name)
[  860s]         try:
[  860s]             return func(*args, **kwargs)
[  860s]         except Exception as e:
[  860s] >           raise type(e)(
[  860s]                 f"An error occurred while calling the {funcname(func)} "
[  860s]                 f"method registered to the {self.backend} backend.\n"
[  860s]                 f"Original Message: {e}"
[  860s]             ) from e
[  860s] E           RuntimeError: An error occurred while calling the read_parquet method registered to the pandas backend.
[  860s] E           Original Message: `pyarrow` not installed
[  860s] 
[  860s] /usr/lib/python3.10/site-packages/dask/backends.py:135: RuntimeError
```

Reply:
Ah - Thanks! This shouldn't try to use pyarrow at all when pyarrow isn't installed. My mistake.